services/ml/compilation_delegate_dnnl.cc

```diff
@@ -0,0 +1,1764 @@
+    return mojom::BAD_DATA;
+  }
+  return mojom::NOT_ERROR;
+}
+
+int32_t CompilationDelegateDnnl::DnnlGetDims(
+    const std::vector<uint32_t>& dimensions,
+    std::vector<dnnl_dim_t>& dnnl_dims,
+    dnnl_format_tag_t format) {
+  dnnl_dims.resize(dimensions.size());
+  if (dimensions.size() == 1) {
+    dnnl_dims[0] = dimensions[0];
+  } else if (dimensions.size() == 2) {
+    dnnl_dims[0] = dimensions[0];
+    dnnl_dims[1] = dimensions[1];
+  } else if (dimensions.size() == 3) {
+    // dnnl logical dimensions come in the order: (n, c, w)
+    // WebNN order is nwc
+    dnnl_dims[0] = dimensions[0];
+    dnnl_dims[1] = dimensions[2];
+    dnnl_dims[2] = dimensions[1];
+  } else if (dimensions.size() == 4) {
+    if (format == dnnl_hwigo) {
+      // for depthwise weights, come in the order: (g, o, i, h, w)
+      // WebNN order is ihwo where o is the number of filters
+      dnnl_dims.resize(5);
+      dnnl_dims[0] = dimensions[3];
+      dnnl_dims[1] = 1;
+      dnnl_dims[2] = 1;
+      dnnl_dims[3] = dimensions[1];
+      dnnl_dims[4] = dimensions[2];
+    } else {
+      // dnnl logical dimensions come in the order: (n, c, h, w)
+      // WebNN order is nhwc
+      dnnl_dims[0] = dimensions[0];
+      dnnl_dims[1] = dimensions[3];
+      dnnl_dims[2] = dimensions[1];
+      dnnl_dims[3] = dimensions[2];
+    }
+  } else {
+    LOG(ERROR) << "Tensor rank " << dimensions.size() << " is not supproted";
+    return mojom::BAD_DATA;
+  }
+  return mojom::NOT_ERROR;
+}
+
+int32_t CompilationDelegateDnnl::DnnlGetDataType(int32_t type,
+                                                 dnnl_data_type_t* dnnl_type) {
+  if (type == mojom::TENSOR_FLOAT32) {
+    *dnnl_type = dnnl_f32;
+  } else if (type == mojom::TENSOR_INT32) {
+    *dnnl_type = dnnl_s32;
+  } else {
+    LOG(ERROR) << "Type " << type << " is not supported";
+    return mojom::BAD_DATA;
+  }
+  return mojom::NOT_ERROR;
+}
+
+int32_t CompilationDelegateDnnl::DnnlCreateMemoryByQueryType(
+    const dnnl_primitive_desc_t& pd,
+    dnnl_query_t query_type,
+    dnnl_memory_t& output_memory) {
+  const dnnl_memory_desc_t* output_desc =
+      LATE(dnnl_primitive_desc_query_md)(pd, query_type, 0);
+  dnnl_status_t status = LATE(dnnl_memory_create)(
+      &output_memory, output_desc, compiled_model_->engine, NULL);
+  if (status != dnnl_success) {
+    LOG(ERROR) << "[DNNL] failed to create memory " << status;
+    return mojom::OP_FAILED;
+  }
+  size_t buffer_size = LATE(dnnl_memory_desc_get_size)(output_desc);
+  void* buffer = base::AlignedAlloc(buffer_size, ALIGNMENT);
+  status = LATE(dnnl_memory_set_data_handle)(output_memory, buffer);
+  if (status != dnnl_success) {
+    LOG(ERROR) << "[DNNL] failed to set data handle to memory primitive "
+               << status;
+    LATE(dnnl_memory_destroy)(output_memory);
+    base::AlignedFree(buffer);
+    return mojom::OP_FAILED;
+  }
+  return mojom::NOT_ERROR;
+}
+
+int32_t CompilationDelegateDnnl::DnnlCreateMemoryDescriptor(
```

Comment:
I think it's unnecessary to prefix `Dnnl` before method such as `DnnlCreateMemoryDescriptor` -> `CreateMemoryDescriptor`

Reply:
Yes, I have remove the unnecessary prefix for some method name.